src/sync_protocol/src/agent_sync_protocol.cpp

```diff
@@ -102,40 +124,44 @@ bool AgentSyncProtocol::sendStartAndWaitAck(const std::string& module, Mode mode
     const size_t buffer_size = builder.GetSize();
     std::vector<uint8_t> messageVector(buffer_ptr, buffer_ptr + buffer_size);
 
-    const auto retryInterval = std::chrono::seconds(30);
-
-    for (int attempt = 1; ; ++attempt)
+    for (;;)
     {
+        if (std::chrono::steady_clock::now() >= deadLine)
+        {
+            std::cerr << "[Sync] Global timeout reached while waiting for StartAck.\n";
+            return false;
+        }
+
         {
```

Comment:
Maybe this block can be moved before the for?

Reply:
Done!